packages/manager/apps/pci-cold-archive/src/pages/containers/new/steps/LinkUserCreation.component.tsx

```diff
@@ -74,20 +74,17 @@ export default function LinkUserCreation({
     poll<TUser>({
       fn: () => getUser(projectId, data.id),
       ruleFn: (user: TUser) => user.status === 'ok',
-      onSuccess: ({ value }) => {
-        generateS3Credentials(projectId, value.id)
-          .then((credentials) => {
-            setNewUser({
-              ...value,
-              access: credentials.access,
-              s3Credentials: credentials,
-            });
-          })
-          .finally(() => {
-            invalidateGetUsersCache(projectId);
-            onCreateUser(value);
-            setIsLoading(false);
-          });
+      onSuccess: async ({ value }) => {
+        const credentials = await generateS3Credentials(projectId, value.id);
```

Comment:
You cannot invoke `generateS3Credentials` with `projectId` that could be `undefined`

Reply:
yes, but in terms of TS typing, it should not be the case ;)